EducationPortal.DAL/FileStorage.Core/Infrastructure/Storage.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using EducationPortal.DAL.Entities;
+using EducationPortal.DAL.FileStorage.Core.Infrastructure.Interfaces;
+
+namespace EducationPortal.DAL.FileStorage.Core.Infrastructure
+{
+    public class Storage : IDatabase
+    {
+        private readonly FSContext context;
+        private readonly string name;
+
+        public Storage(FSContext context)
+        {
+            this.context = context;
+            this.name = typeof(Storage).Namespace;
+            // this.name = Assembly.GetExecutingAssembly().FullName;
+
+            // this.EnsureCreated();
+        }
+
+        // true if created, false if it already existed
+        public virtual bool EnsureCreated()
+        {
+            if (!this.Exists())
```

Comment:
if not exist it returns false?

Reply:
Creates storage if it not exists and return true/
If storage already exists method returns false